_sources/vectors_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,292 @@
+    =====
+    int main() {
+    =====
+        vector<string> nouns = {"cereal", "Cocoa Puffs", "Mario", "luigi", "Aerosmith"};
+    =====
+        vector<string> proper = {};
+    =====
+        for (int i = 0; i < nouns.size(); i++) {
+    =====
+            if (isupper(nouns[i][0])) {
+    =====
+                proper.push_back(nouns[i]);
+    =====
+            }
+    =====
+        }
+    =====
+    }
+    =====
+    proper.push_back(nouns[i][0]); #distractor
+    =====
+    proper.pop_back(nouns[i]); #distractor
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_6
+    :numbered: left
+    :adaptive:
+
+    Suppose you have the following function ``howMany`` and vector ``exclamations``
+    Construct a block of code that counts how many times ".", "!", and "?" occur in ``exclamations``.
+    Save the counts to a vector with "." count as the first element, "!" count as the second, and "?" count as the third.
+    Put the necessary blocks of code in the correct order.
+
+    ::
+
+        int howMany (const vector<string>& vec, char let) {
+            int count = 0;
+            for (int i = 0; i < vec.size(); i++) {
+                for (int k = 0; k < vec[i].size(); k++) {
+                    if (vec[i][k] == let) {
+                        count++;
+                    }
+                }
+            }
+            return count;
+        }
+
+        vector<string> exclamations = {"what?!", "how???", "fine!", "STOP.", "yay!!!!!", "ugh...!"};
+
+    -----
+    #include <vector>
+    #include <string>
+    using namespace std;
+    =====
+    int main() {
+    =====
+        vector<string> exclamations = {"what?!", "how???", "fine!", "STOP.", "yay!!!!!", "ugh...!"};
+    =====
+        vector<char> punc = {'.', '!', '?'};
+    =====
+        vector<int> counts = {};
+    =====
+        for (int i = 0; i < punc.size(); i++) {
+    =====
+            counts.push_back(howMany(excl, punc[i]));
+    =====
+        }
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_7
+    :numbered: left
+    :adaptive:
+
+    Write the function ``endsEven`` that takes a vector and removes elements from the end of the vector until
+    it ends with an even number. Put the necessary blocks of code in the correct order.
+    -----
+    void endsEven (vector<int> &vec) {
+    =====
+    vector endsEven (vector<int> &vec) { #distractor
+    =====
+        while (vec.back() % 2 != 0) {
+    =====
+        for (int i = 0; i < vec.size(); i++) { #paired
+    =====
+        vec.pop_back();
+    =====
+        }
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_8
+    :numbered: left
+    :adaptive:
+
+    Write the function ``randomNums`` that takes two integers: ``num`` which is the number of random numbers
+    you wish to generate, and ``max``, which is the maximum value of random number you wish to generate.  Your
+    function should return a vector of ``num`` integers that are between 1 and ``max``, inclusive.
```

Comment:
Because we teach students to always pass vectors by reference or const reference, I think it would be best to re-write this function so it took randomNums as a parameter by reference and then filled it with random numbers